packages/pegjs/lib/compiler/passes/generate-js.js

```diff
@@ -786,6 +790,12 @@ function generateJS( ast, session, options ) {
                         ip++;
                         break;
 
+                    case op.PUSH_FATAL:         // PUSH_FATAL
+                        stack.push( "peg$FAILED ");
```

Comment:
Why is there a space character at the end of the string?

Reply:
because there's more text after it, and so the space creates a gap between words